hookman/hookman_generator.py

```diff
@@ -106,7 +111,58 @@ def format_arguments(text_list: List[str]) -> str:
                     function_name=f'{self.project_name}_{self.version}_{hook_spec.__name__.lower()}',
                 ))
 
-    def generate_files(self, dst_path: Path):
+    def generate_plugin_template(self,
+            plugin_name: str,
+            shared_lib_name: str,
+            author_email: str,
+            author_name: str,
+            dst_path: Path):
+        """
+        Generate a template with the necessary files and structure to create a plugin
+            - config.yml
+            - plugin.c
+            - hook_specs.h
+            - CMakeLists.txt
+            - README.md
+        """
+        plugin_folder = dst_path / plugin_name
+        assets_folder = plugin_folder / 'assets'
+        source_folder = plugin_folder / 'src'
+
+        if not plugin_folder.exists():
+            plugin_folder.mkdir(parents=True)
+
+        if not assets_folder.exists():
+            assets_folder.mkdir()
+
+        if not source_folder.exists():
+            source_folder.mkdir()
+
+        build_script_file = Path(plugin_folder / 'build.py')
```

Comment:
This is a matter of style, but I think creating that many local variables make the code harder to follow.

For example:

```python
plugin_cmake_file = Path(plugin_folder / 'CMakeLists.txt')
...
plugin_cmake_content = self._plugin_cmake_file_content(shared_lib_name)
...
plugin_cmake_file.write_text(plugin_cmake_content)
```

I find it easier to read and spot errors when written like:

```python
Path(plugin_folder / 'CMakeLists.txt').write_text(
    self._plugin_cmake_file_content(shared_lib_name))
```

Because this is much less variables to track and keep on your head. Also easier to spot mistakes.

But if you prefer it like this please do so, as I said is a matter of style.

Reply:
Actually, I prefer this way too, I just thought that inserting too many statements in one line would be too "cpp like" and less "pythonic "

Reply:
- [x] Fixed